### ISSUE TITLE
Pivot: all PivotItems render always (alwaysRender broken)

### ISSUE THREAD

**Issue by OneCyrus** [microsoft/fluentui#16126]
### Environment Information

- **Package version(s)**: 7.153.3
- **Browser and OS versions**: Chrome 87

### Please provide a reproduction of the bug in a codepen:

https://codepen.io/OneCyrus/pen/abmvrjz

#### Actual behavior:

All PivotItems render on initialization. Even though alwaysRender should be false per default value. (you can verify this in the console of the codepen repro)

![image](https://user-images.githubusercontent.com/7747800/100978202-58459180-3542-11eb-8e92-d05f077b1a8e.png)
![image](https://user-images.githubusercontent.com/7747800/100978234-6693ad80-3542-11eb-98f8-482944b2038e.png)

#### Expected behavior:

Only the selected PivotItem should get rendered.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: High (high performance regression in our app)

Products/sites affected: internal product

**Comment by JustSlone** [comment]
@OneCyrus thanks for reaching out!

Actually it looks like this is due to how react evaluates statements inside {}. It's evaluating that statement as part of creating the PivotItem element. Passing it into PivotItem as children, so it gets run before PivotItem is called. 

However, since react components turn into a call to React.createElement, this doesn't happen for components. So the components in-fact are not getting rendered. We can confirm this by adding a simple <Logger> component to the example that performs the console.log on render. Here's an example: [16126 - example - PivotItem logging](https://codepen.io/justslone-the-encoder/pen/RwGagPp?editors=0011)

To demonstrate this with code:
Basically this JSX 
```jsx
<PivotItem headerText="Recent" itemKey="recent" alwaysRender={false}>
   {console.log("Immedite #2")}
   <Logger text="Recent" />
   <Label styles={labelStyles}>Pivot #2</Label>
</PivotItem>
```

turns into this JavaScript
```js
React.createElement(
PivotItem, { headerText: "Recent", itemKey: "recent", alwaysRender: false },
   console.log("Immedite #2"),
   React.createElement(Logger, { text: "Recent" }),
   React.createElement(Label, { styles: labelStyles }, "Pivot #2")
)
```

I hope this answers your question!

**Comment by OneCyrus** [comment]
actually we hit this issue with components inside the pivotitem. e.g. we have apollo queries inside the pivotitems and all requests fire at the same time. something seems to be strange here.

**Comment by JustSlone** [comment]
Hmm, that is strange. Can you share a CodePen with that behavior? 

As long as the calls are inside the render path, they shouldn't be called. Perhaps something is being called on component creation?

**Comment by OneCyrus** [comment]
ok looks like it has something to do with the react.strictmode in development builds. i guess fluentui works correct.
